alphadia/workflow/peptidecentric/peptidecentric.py

```diff
@@ -920,110 +916,29 @@ def log_precursor_df(self, precursor_df):
+            self._requantification_handler = RequantificationHandler(
+                self.config,
+                self.calibration_manager,
+                self.fdr_manager,
+                self.reporter,
+                self.spectral_library,
+                rt_column=self._get_rt_column(),
+                mobility_column=self._get_mobility_column(),
+                precursor_mz_column=self._get_precursor_mz_column(),
+                fragment_mz_column=self._get_fragment_mz_column(),
             )
-        )
-        multiplexed_candidates = multiplex_candidates(
-            reference_candidates,
-            self.spectral_library.precursor_df_unfiltered,
-            channels=channels,
-        )
-
-        channel_count_lib = self.spectral_library.precursor_df_unfiltered[
-            "channel"
-        ].value_counts()
-        channel_count_multiplexed = multiplexed_candidates["channel"].value_counts()
-        ## log channels with less than 100 precursors
-        for channel in channels:
-            if channel not in channel_count_lib:
-                self.reporter.log_string(
-                    f"channel {channel} not found in library", verbosity="warning"
-                )
-            if channel not in channel_count_multiplexed:
-                self.reporter.log_string(
-                    f"channel {channel} could not be mapped to existing IDs.",
-                    verbosity="warning",
-                )
-
-        self.reporter.log_string(
-            f"=== Requantifying {len(multiplexed_candidates):,} precursors ===",
-            verbosity="progress",
-        )
-
-        config = CandidateConfig()
-        config.score_grouped = True
-        config.exclude_shared_ions = True
-        config.reference_channel = self.config["multiplexing"]["reference_channel"]
-        config.experimental_xic = self.config["search"]["experimental_xic"]
-
-        multiplexed_scoring = CandidateScoring(
-            self.dia_data.jitclass(),
-            self.spectral_library.precursor_df_unfiltered,
-            self.spectral_library.fragment_df,
-            config=config,
-            rt_column=self._get_rt_column(),
-            mobility_column=self._get_mobility_column(),
-            precursor_mz_column=self._get_precursor_mz_column(),
-            fragment_mz_column=self._get_fragment_mz_column(),
-        )
-
-        multiplexed_candidates["rank"] = 0
 
-        multiplexed_features, fragments = multiplexed_scoring(multiplexed_candidates)
+    def requantify(self, psm_df: pd.DataFrame) -> pd.DataFrame:
+        """TODO.
 
-        target_channels = [
-            int(c) for c in self.config["multiplexing"]["target_channels"].split(",")
-        ]
-        reference_channel = self.config["multiplexing"]["reference_channel"]
+        Delegates to RequantificationHandler.requantify(), see docstring there for more details.
+        """
+        self._lazy_init_requantification_handler()
```

Comment:
can't we move this code here? I'm not sure how much we are winning by isolating it into a function